includes/index_filters/default_filter.php

```diff
@@ -19,168 +20,149 @@
   $_GET['sort'] = substr($_GET['sort'], 0, 3);
 }
 if (isset($_GET['alpha_filter_id']) && (int)$_GET['alpha_filter_id'] > 0) {
-    $alpha_sort = " and pd.products_name LIKE '" . chr((int)$_GET['alpha_filter_id']) . "%' ";
+  $alpha_sort = " AND pd.products_name LIKE '" . chr((int)$_GET['alpha_filter_id']) . "%' ";
+} else {
+  $alpha_sort = '';
+}
+if (!isset($select_column_list)) {
+  $select_column_list = '';
+}
+if (!isset($do_filter_list)) {
+  $do_filter_list = false;
+}
+$and = '';
+// show the products of a specified manufacturer
+if (isset($_GET['manufacturers_id']) && $_GET['manufacturers_id'] != '') {
+  if (isset($_GET['filter_id']) && zen_not_null($_GET['filter_id'])) {
+// We are asked to show only a specific category
+    $and = " AND m.manufacturers_id = " . (int)$_GET['manufacturers_id'] . "
+            AND p2c.categories_id = " . (int)$_GET['filter_id'] . " ";
   } else {
-    $alpha_sort = '';
-  }
-  if (!isset($select_column_list)) $select_column_list = "";
-   // show the products of a specified manufacturer
-  if (!isset($do_filter_list)) {
-    $do_filter_list = false;
+    // We show them all
+    $and = " AND m.manufacturers_id = " . (int)$_GET['manufacturers_id'] . " ";

```

Comment:
This:
````php
 if (isset($_GET['filter_id']) && zen_not_null($_GET['filter_id'])) {
// We are asked to show only a specific category
    $and = " AND m.manufacturers_id = " . (int)$_GET['manufacturers_id'] . "
            AND p2c.categories_id = " . (int)$_GET['filter_id'] . " ";
  } else {
    $and = " AND m.manufacturers_id = " . (int)$_GET['manufacturers_id'] . " ";
  }
````
Looks like it could be shortened to:
````php
 $and .= " AND m.manufacturers_id = " . (int)$_GET['manufacturers_id'] . " ";
 if (isset($_GET['filter_id']) && zen_not_null($_GET['filter_id'])) {
// We are asked to show only a specific category
    $and .= "AND p2c.categories_id = " . (int)$_GET['filter_id'] . " ";
  }
````
With a similar revision to the category section below or is there a need to keep these two equivalent sections within the `if/else` separate?

Overall it certainly looks better (easier to follow) than what it used to look like.

Reply:
I think your revision looks great. I'll go test it, and see if I can shorten the if/else.
My main goal was loosing the quadruple queries.